src/Typesense/Documents.ts

```diff
@@ -0,0 +1,240 @@
+import ApiCall from './ApiCall'
+import Collections from './Collections'
+import Configuration from './Configuration'
+import RequestWithCache from './RequestWithCache'
+import { ImportError } from './Errors'
+
+// Todo: use generic to extract filter_by values
+export interface DeleteQuery {
+  filter_by: string
+  batch_size?: number
+}
+
+export interface DeleteResponse {
+  num_deleted: number
+}
+
+interface ImportResponseSuccess {
+  success: true
+}
+interface ImportResponseFail {
+  success: false
+  error: string
+  document: DocumentSchema
+  code: number
+}
+export type ImportResponse = ImportResponseSuccess | ImportResponseFail
+
+export interface DocumentSchema extends Record<string, any> {
+  // id?: string //may actually give trouble if someone uses non-string id's
+}
+
+export interface SearchParams<T extends DocumentSchema> {
+  // From https://typesense.org/docs/latest/api/documents.html#arguments
+  q: string
+  query_by: string
+  query_by_weights?: string
+  prefix?: boolean // default: true
+  filter_by?: string
+  sort_by?: string // default: text match desc
+  facet_by?: string
+  max_facet_values?: number
+  facet_query?: string
+  page?: number // default: 1
+  per_page?: number // default: 10, max 250
+  group_by?: keyof T
+  group_limit?: number // default:
+  include_fields?: string
+  exclude_fields?: string
+  highlight_fields?: string // default: all queried fields
+  highlight_full_fields?: string // default: all fields
+  highlight_affix_num_tokens?: number // default: 4
+  highlight_start_tag?: string // default: <mark>
+  highlight_end_tag?: string // default: </mark>
+  snippet_threshold?: number // default: 30
+  num_typos?: string // default: 2
+  drop_tokens_threshold?: number // default: 10
+  typo_tokens_threshold?: number // default: 100
+  pinned_hits?: string
+  hidden_hits?: string
+  limit_hits?: number // default: no limit
+  pre_segmented_query?: boolean
+  enable_overrides?: boolean
+  prioritize_exact_match?: boolean // default: true
+  [key: string]: any // allow for future parameters without having to update the library
```

Comment:
This is somewhat controversial because it won't flag the user if they make a typo in the property name. It will still suggest valid options, however `prioritize_exact_match` and `proritize_exact_match` are both valid even though the latter is a typo.